structural/adapter/PaymentAdapter.js

```diff
@@ -0,0 +1,16 @@
+class PaymentAdapter {
```

Comment:
I'd suggest to show off the inheritance here:
PaymentAdapter extends  ExistingPaymentProcessor
So that we explicitly specify that this class could be used instead of the ExistingPaymentProcessor in any scenario